src/devices_models/devices/thermal_generation.jl

```diff
@@ -150,6 +150,19 @@ function activepower_constraints!(
         variable_name(ACTIVE_POWER, T),
         variable_name(ON, T),
     )
+    for (ix, d) in enumerate(devices)
+        name = PSY.get_name(d)
+        limits = PSY.get_activepowerlimits(d)
+        range_data = DeviceRange(name, limits)
+        add_device_services!(range_data, d, model; add_supplemental=true)
+        constraint_data[ix] = range_data
+    end
+    device_range(
+        psi_container,
+        constraint_data,
+        constraint_name(ACTIVE_RANGE, T),
+        variable_name(ACTIVE_POWER, T),
+    )
```

Comment:
For my edification, can you explain what this function call is doing?

Given its name suggest it's _not_ a mutating function, and we don't return the result, it looks as if it does nothing... but the same can be said of the exisiting call to `device_semicontinuousrange` above.

Looking at the code for `device_range` https://github.com/NREL-SIIP/PowerSimulations.jl/blob/d949d73564b1a91fed5d9291d28fab122e9f5cf9/src/devices_models/devices/common/range_constraint.jl#L43-L44
it seems it actually _is_ a mutating function (which would explain its. usage here) - is that right? If so, why is it not named `device_range!` with the trailing `!` to indicate it mutates the `PSIContainer`?

Reply:
You're right -- see https://github.com/NREL-SIIP/PowerSimulations.jl/issues/485 for clarification. The naming is being changed in the next release to avoid confusion.

Reply:
Oh you're ahead of the game - thank you!